File: 36-valid-sudoku/valid-sudoku.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        // Create 9 sets for rows, columns, and blocks
        Set<Character>[] Row = new HashSet[9];
        Set<Character>[] Col = new HashSet[9];
        Set<Character>[] Block = new HashSet[9];

        for (int i = 0; i < 9; i++) {
            Row[i] = new HashSet<>();
            Col[i] = new HashSet<>();
            Block[i] = new HashSet<>();
        }

        // Traverse the board
        for (int i = 0; i < 9; i++)
            for (int j = 0; j < 9; j++) {
                char c = board[i][j];
                if (c == '.') continue;

                // ✅ Check row
                if (Row[i].contains(c)) return false;
                Row[i].add(c);

                // ✅ Check column
                if (Col[j].contains(c)) return false;
                Col[j].add(c);

                // ✅ Check block
                int idx = (i / 3) * 3 + (j / 3);
                if (Block[idx].contains(c)) return false;
                Block[idx].add(c);
            }

        return true;
    }
}

File: 36-valid-sudoku/valid-sudoku.go
func isValidSudoku(board [][]byte) bool {
    Row := make([]map[byte]bool, 9)
    Col := make([]map[byte]bool, 9)
    Block := make([]map[byte]bool, 9)
    
    for i := 0; i < 9; i++ {
        Row[i] = make(map[byte]bool)
        Col[i] = make(map[byte]bool)
        Block[i] = make(map[byte]bool)
    }

    for i := 0; i < 9; i++ {
        for j := 0; j < 9; j++ {
            c := board[i][j]
            if c == '.' {
                continue
            }

            // ✅ Check row
            if Row[i][c] {
                return false
            }
            Row[i][c] = true

            // ✅ Check column
            if Col[j][c] {
                return false
            }
            Col[j][c] = true

            // ✅ Check block
            idx := (i/3)*3 + (j/3)
            if Block[idx][c] {
                return false
            }
            Block[idx][c] = true
        }
    }

    return true
}